staging/src/k8s.io/kubectl/pkg/drain/drain_test.go

```diff
@@ -220,90 +220,114 @@ func TestCheckEvictionSupport(t *testing.T) {
 
-						if evictionSupported {
-							eviction := policyv1beta1.Eviction{}
-							eviction.Kind = "Eviction"
-							eviction.APIVersion = "policy/v1"
-							eviction.Namespace = pod.Namespace
-							eviction.Name = pod.Name
+				create = append(create, pod)
+				if i <= 2 {
+					deletePods = append(deletePods, *pod)
 
-							gracePeriodSeconds := int64(h.GracePeriodSeconds)
-							eviction.DeleteOptions = &metav1.DeleteOptions{
-								GracePeriodSeconds: &gracePeriodSeconds,
-							}
+					if tc.evictionSupported && !tc.disableEviction {
+						eviction := policyv1beta1.Eviction{}
+						eviction.Kind = "Eviction"
+						eviction.APIVersion = "policy/v1"
+						eviction.Namespace = pod.Namespace
+						eviction.Name = pod.Name
 
-							expectedEvictions = append(expectedEvictions, eviction)
+						gracePeriodSeconds := int64(h.GracePeriodSeconds)
+						eviction.DeleteOptions = &metav1.DeleteOptions{
+							GracePeriodSeconds: &gracePeriodSeconds,
 						}
+
+						expectedEvictions = append(expectedEvictions, eviction)
 					}
 				}
+			}
 
-				// Build the fake client
-				k := fake.NewSimpleClientset(create...)
-				if evictionSupported {
-					addEvictionSupport(t, k)
-				}
-				h.Client = k
+			// Build the fake client
+			k := fake.NewSimpleClientset(create...)
+			if tc.evictionSupported {
+				addEvictionSupport(t, k)
+			}
+			h.Client = k
+			h.DisableEviction = tc.disableEviction
+			// Do the eviction
+			if err := h.DeleteOrEvictPods(deletePods); err != nil {
+				t.Fatalf("error from DeleteOrEvictPods: %v", err)
+			}
 
-				// Do the eviction
-				if err := h.DeleteOrEvictPods(deletePods); err != nil {
-					t.Fatalf("error from DeleteOrEvictPods: %v", err)
+			// Test that other pods are still there
+			var remainingPods []string
+			{
+				podList, err := k.CoreV1().Pods("").List(metav1.ListOptions{})
+				if err != nil {
+					t.Fatalf("error listing pods: %v", err)
 				}
 
-				// Test that other pods are still there
-				var remainingPods []string
-				{
-					podList, err := k.CoreV1().Pods("").List(metav1.ListOptions{})
-					if err != nil {
-						t.Fatalf("error listing pods: %v", err)
-					}
-
-					for _, pod := range podList.Items {
-						remainingPods = append(remainingPods, pod.Namespace+"/"+pod.Name)
-					}
-					sort.Strings(remainingPods)
-				}
-				expected := []string{"default/mypod-3", "default/mypod-4"}
-				if !reflect.DeepEqual(remainingPods, expected) {
-					t.Errorf("unexpected remaining pods after DeleteOrEvictPods; actual %v; expected %v", remainingPods, expected)
+				for _, pod := range podList.Items {
+					remainingPods = append(remainingPods, pod.Namespace+"/"+pod.Name)
 				}
+				sort.Strings(remainingPods)
+			}
+			expected := []string{"default/mypod-3", "default/mypod-4"}
+			if !reflect.DeepEqual(remainingPods, expected) {
+				t.Errorf("%s: unexpected remaining pods after DeleteOrEvictPods; actual %v; expected %v", tc.description, remainingPods, expected)
```

Comment:
Nit. The description shouldn't be needed here. Since each example is a separate subtest the framework will include this information for test failures.

Reply:
I felt it was a little redundant, I was just following the existing pattern.  Perhaps I'll cleanup the package in a follow up.